core/amber/src/main/scala/edu/uci/ics/amber/engine/common/ambermessage/WorkflowMessage.scala

```diff
@@ -0,0 +1,3 @@
+package edu.uci.ics.amber.engine.common.ambermessage
+
+class WorkflowMessage {}
```

Comment:
Now every control message extends WorkflowMessage, which makes things complicated. I suggest we isolate the message and the control/data so that the workflow message carries control/data event. i.e. When we send control/data, we use a control/data event and then pass it to the messaging layer. The messaging layer will wrap this event and make it a WorkflowMessage with the sequence number and everything. When this message comes to the receiver end, the FIFO enforcer will unwrap it and pass the event to the internal logic.

Reply:
I also suggest we create a diagram to describe how the message flows inside the messaging layer.

Reply:
I agree with the broad picture you are suggesting. Let's discuss